tests/io/v0_0_0/test_http_backend.py

```diff
@@ -0,0 +1,145 @@
+import contextlib
+import json
+import os
+import socket
+import subprocess
+import sys
+import time
+import unittest
+
+import numpy as np
+import requests
+import skimage.io
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", ".."))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+import slicedimage
+from tests.utils import TemporaryDirectory
+
+
+class TestHttpBackend(unittest.TestCase):
+    def setUp(self, timeout_seconds=5):
+        self.contexts = []
+        self.tempdir = TemporaryDirectory()
+        self.contexts.append(self.tempdir)
+        self.port = _unused_tcp_port()
+
+        if sys.version_info[0] == 2:
+            module = "SimpleHTTPServer"
+        elif sys.version_info[0] == 3:
+            module = "http.server"
+        else:
+            raise Exception("unknown python version")
+
+        self.contexts.append(ContextualChildProcess(
+            [
+                "python",
+                "-m",
+                module,
+                str(self.port),
+            ],
+            cwd=self.tempdir.name,
+        ).__enter__())
+
+        end = time.time() + timeout_seconds
+        while True:
+            try:
+                requests.get("http://0.0.0.0:{port}".format(port=self.port))
+                break
+            except requests.ConnectionError:
+                if time.time() > end:
+                    raise
+
+    def tearDown(self):
+        for context in self.contexts:
+            context.__exit__(*sys.exc_info())
+
+    def test_tiff(self):
+        """
+        Generate a tileset consisting of a single TIFF tile.  Deposit it where the HTTP server can
+        find the tileset, and fetch it.
+        """
+        # write the tiff file
+        data = np.random.randint(0, 65535, size=(100, 100), dtype=np.uint16)
+        skimage.io.imsave(os.path.join(self.tempdir.name, "tile.tiff"), data, plugin="tifffile")
+
+        # TODO: (ttung) We should really be producing a tileset programmatically and writing it
```

Comment:
This TODO seems inconsistent with the tiffwriter above.

Reply:
See https://github.com/spacetx/slicedimage/pull/22/files#diff-e1f72ba855d4777b00c47a4703013bbfR104 for how it should probably be done.